### ISSUE TITLE
macOS 10.12: Audio always stuttering and skipping on MacBookAir6,2

### ISSUE THREAD

**Issue by ghost** [mumble-voip/mumble#2624]
On the 2013 MacBook Air 13" with macOS 10.12 Sierra, my sent and received audio with the default settings sounds as if it is losing many packets. The info window still reports 0% lost or late packets, even when people talk for whole minutes, so I don't think it's a network problem, and could be related to some internal changes in macOS. This issue is not present in OS X 10.9.5 Mavericks, which I installed as a test, and all audio works as expected with no problems in 10.9.5. I could not find any settings in Mumble 1.2.17 or macOS able to solve this issue.

I thought this was another Broadcom WLAN issue, but it appears not to be. Card is AirPort extreme 
(0x14E4, 0x117) with a BCM43xx chip (System Profiler doesn't say which in particular.) I'm connected over 5Ghz Wireless-n and have a 90Mbit broadband connection. Switching to 2.4Ghz mode makes no difference.

**Comment by Kissaki** [comment]
> I could not find any settings in Mumble 1.2.17 or macOS able to solve this issue.

Just to make sure, this includes increasing the Jitter buffer?

You said this is also for sent audio. Other people report this to you?

**Comment by BoctorPoo** [comment]
> Just to make sure, this includes increasing the Jitter buffer?
Yes, I tried changing that one the most, to no avail.
> You said this is also for sent audio. Other people report this to you?
Yes, when I speak, it skips a bit and has "robotic" artifacts. Sometimes, my speech even loops the same buffer again and again until I release my Push-To-Talk key. When it happens, everyone connected hears it and usually comments on it, asking me to repeat my entire sentence because of the problem.

**Comment by mkrautz** [comment]
FWIW, I'm not able to reproduce on macOS Sierra using a MacBookPro11,3.

So maybe not related to audio changes in Sierra?

**Comment by mkrautz** [comment]
Oh, this was the latest snapshot build, 1.3.0~1650~g30c779c~snapshot.

**Comment by mkrautz** [comment]
(I have yet to test on a stable release. Sorry.)

**Comment by BoctorPoo** [comment]
> Oh, this was the latest snapshot build, 1.3.0~1650~g30c779c~snapshot.

Just gave that snapshot a try on my MacBookAir6,2 with macOS 10.12.1, and unfortunately ran into the same problems. Yet on the same computer, under OS X 10.9.5, this version still works as expected without stuttering or repeating bits of my speech to other clients.
I've begun wondering if Apple messed up a Broadcom driver or kext between 2014 and 2015, as I remember the problem being nonexistent on Yosemite, but manifesting itself by the time El Capitan was out. The problem is completely identical on my Lenovo T420 which can unofficially run OS X, and it's using a BCM4322 WLAN card.
My last piece of Apple hardware to test with is a MacBook Pro from 2007, but the last OS X version it can run is El Capitan. I'm thinking it may have different results, since it has a completely different WLAN card.

**Comment by BoctorPoo** [comment]
Small update. I have tested on a 2015 15" MacBook Pro (MacBookPro11,4) and the issue is surprisingly not there. However, the Airport Extreme WLAN card in this model– still a BCM43xx– reports a vendor/model of 0x14E4, 0x152, as opposed to the 0x117 in the 2013 MacBook Air.
Both stable 1.2.17 and the current snapshot at this time work fine on El Capitan and Sierra on the MacBook Pro, despite neither working properly on the Air.

**Comment by mkrautz** [comment]
So on the Pro, it's harder to trigger? Anything that can trigger it easily?

And on the Air, it happens all the time?

**Comment by BoctorPoo** [comment]
On the Pro, it's slightly less common, but it will happen all the time on either of them, seemingly if there's not enough activity happening on the computer. It's seemingly related to CPU states, and fiddling around in Safari is sometimes enough to get the stutters to stop for a while.

**Comment by Krzmbrzl** [comment]
Is this bug still in Mumble 1.3?

**Comment by HurkWurk** [comment]
This is indeed still an issue in 1.3. a friend uses a mac and i hear the looping/skipping mentioned here. he updated mumble both stable and dev and neither helped.  hes using catalina 15.3 if i recall.

**Comment by davidebeatrici** [comment]
Does it happen with other applications as well?

**Comment by HurkWurk** [comment]
yes, im not personally very familiar with the issue, but here is more on
it:
https://appleinsider.com/articles/19/02/19/pro-audio-glitch-with-t2-equipped-macs-connected-to-usb-20-connections

**Comment by tparikka** [comment]
I'm wondering if folks here could sign off with what they are listening on: speakers, AirPods, hardwired headphones, other? There was a mention in #6680 of audio issues only presenting over Bluetooth.

**Comment by tparikka** [comment]
I have been experimenting on this issue. I have not been able to reproduce the problem with USB headphones or with Mumble on speakers/mic. I'll be trying more with my AirPods soon to see if the issue is still reproducible with those.

**Comment by tparikka** [comment]
I am still testing with this issue.
